src/Sylius/OrderProcessing/OrderFeeProcessor.php

```diff
@@ -146,39 +136,29 @@ public function process(BaseOrderInterface $order): void
             }
         }
 
-        $tipAmount = $order->getTipAmount();
-
-        if (null !== $tipAmount) {
-            if ($tipAmount > 0) {
-                $tipAdjustment = $this->adjustmentFactory->createWithData(
-                    AdjustmentInterface::TIP_ADJUSTMENT,
-                    $this->translator->trans('order.adjustment_type.tip'),
-                    $tipAmount,
-                    $neutral = false
-                );
-                $order->addAdjustment($tipAdjustment);
-
-                $this->logger->info(sprintf('Order %s | OrderFeeProcessor | tip amount: %d | added',
-                    $this->loggingUtils->getOrderId($order), $tipAmount));
-            }
-        } else {
-            if (count($originalTipAdjustments) > 0) {
-                foreach ($originalTipAdjustments as $tipAdjustment) {
-                    $order->addAdjustment($tipAdjustment);
-
-                    $this->logger->info(sprintf('Order %s | OrderFeeProcessor | tip amount: %d | re-used previous tip adjustment',
-                        $this->loggingUtils->getOrderId($order), $tipAdjustment->getAmount()));
-                }
-            }
+        $newTipAmount = $order->getTipAmount();
+
+        // $order->getTipAmount() is the tip amount set in the current request
+        // preserve the previous tip amount if it's null
```

Comment:
"if the new tip amount is null/not set" 

("it" could mean "the old tip amount")